frontend/src/components/Combat/Combat.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import * as Colyseus from 'colyseus.js';
+
+interface Props {}
+interface State {
+    roomInstance: Colyseus.Room;
+    lastMove: string | undefined;
+    loading: boolean;
+}
+
+export default class Combat extends Component<Props, State> {
```

Comment:
I felt better working with a class component here since I have explicit access to lifecycle methods